### ISSUE TITLE
[Feedback wanted]: Grouping and reorganising our graphs

### ISSUE THREAD

**Issue by Alportan** [electricitymaps/electricitymaps-contrib#7022]
## Description

We are currently looking into revising our graphs when it comes to their behaviour and data points. 

We have a hypothesis that by combining graphs that show similar data points, and organising it into a cohesive narrative, we can enhance the clarity of our app. 

Here's a glimpse of how that would look like when combining our 2 graphs that show the sources of electricity today (Electricity consumption by source + Origin of electricity)

![image](https://github.com/user-attachments/assets/6d443ecd-9b31-439d-92c4-30aeb3ddd0bf)

As you can see, in this version of the consumption breakdown view (per source), we can add more details and plot more data, in this example the carbon intensity of each source.

Following the same logic, we would bundle our Net exchange graph together with the detailed exchanges we currently shown inside the Electricity Consumption by Source graph.

![image](https://github.com/user-attachments/assets/5d8e8854-6a6d-465e-aead-da6c8286c289)

We've also added a little insight into how did exports changed, in comparison with the past 12 months average.

What we are looking for:
- would this help you draw better conclusions using our data?
- would an insight for each graph bring more value? if yes, which insights do you find most valuable?
- is there anything else you spotted with this approach?

Looking forward to hearing your thoughts 🙏

**Comment by phiphou** [comment]
Love the global imported/exported overview !

**Comment by jayaddison** [comment]
The selected-duration charts are useful, and I like the idea of grouping those logically above the detailed info tables - presenting the trends is a good way to help people get a sense for regional energy data.

I was going to ask whether the plan would still be to have two tabs, one for consumption data and one for emissions data -- but I think I might have answered that while writing this: it appears that the idea is to use the same brown-to-green emissions colour scale to add a dot for each row in the consumption info table?

If so, I like that idea too -- fewer tabs/selections generally makes information more discoverable.  The table is informationally dense, but provided that it uses consistent styling with the rest of application, I think that works.

**Comment by thomasgibon** [comment]
I may be a bit late, and not sure how technically feasible this is, but I think a nice addition would be the composition of imported/exported electricity. This could also be represented in "Origin of electricity", where the bar for each country would be split into sources instead of one bar with a brown-green shade (consistently with domestic production).

This is some test I've made, representing the consumption mix for an average week in 2023 in Germany and Denmark:
![image](https://github.com/user-attachments/assets/5b402c30-c0b7-4f5d-a221-5ff8ace3235f)

I think the "nature" of imported electricity is as important at its carbon content, and knowing this provides some insight about which sources are the most likely to be imported (= which sources are key for the grid stability).

**Comment by Alportan** [comment]
Hi @thomasgibon,

As Viktor mentioned, we've already put some thought into this idea. Here are our design suggestions for presenting 👇 

![image](https://github.com/user-attachments/assets/a88e87cc-4f48-42c6-b0bc-d70eaece1afb)

Let us know your thoughts.

**Comment by jayaddison** [comment]
Re: imported sources, there are two views of the data that I'd expect people might be interested in:

  * Regardless of origin, how clean is energy generation in a country?
    * This closely matches what the diagram with 'Reveal imported sources' already displays: the stacked bar chart has colour-coded energy by generation method, and for the curious, the partial-transparency of the detailed per-method generation/capacity row-bars indicate how much is from import.
    * I find the `|<-` indicator symbols in the row-bars where some capacity is imported visually noisy in the provided example, because they exist for almost all of the rows.  It _might_ be more useful to indicate that `Coal` and `Gas` in the example are locally-generated -- or perhaps it would be easier to omit these indicator symbols entirely?

  * How clean is the energy generation in a country?
    * For this use case, I would prefer imports to be stacked at the bottom of the bar chart, to communicate that although it's a relevant factor, it is not the data exploration focus of the chart.  I think the gray colour indicates that to some extent already.
    * Similarly, I briefly wondered whether putting the imports underneath the x-axis (y=0) could illustrate the way that increased imports allow reduced local production, and that increased local production allows reduction of imports (the chart would appear to push/pull the stack in either direction) -- but both values are positive numbers, so I'm not sure whether this is sensible.

**Comment by Alportan** [comment]
>   * I find the `|<-` indicator symbols in the row-bars where some capacity is imported visually noisy in the provided example, because they exist for almost all of the rows.  It _might_ be more useful to indicate that `Coal` and `Gas` in the example are locally-generated -- or perhaps it would be easier to omit these indicator symbols entirely?

Good point! I like your suggested approach, looking into which sources are locally produced instead of imported. 💡 

>   * [...] I briefly wondered whether putting the imports underneath the x-axis (y=0) could illustrate the way that increased imports allow reduced local production, and that increased local production allows reduction of imports (the chart would appear to push/pull the stack in either direction) -- but both values are positive numbers, so I'm not sure whether this is sensible.

I do agree that it would be useful to find a way to visually represent the relationship between imports and local production. It could be tricky though! 🤔

**Comment by lammel-hub** [comment]
Great thoughts! I would suggest that a simple "rearrange" function is included, so that the order of graphs/groups can be adjusted by the user, e.g. showing spot orice first, the CO2 etc.

**Comment by thomasgibon** [comment]
> Let us know your thoughts.

Hi! I realized I never answered this – I personally really like the "Reveal imported sources" is an important feature. Visually, I would still split domestic vs. imported share in the top chart, and _then_ show the origin of electricity in the domestic mix and imported mix. You could apply transparency to the imported shares, to be consistent with the per-source detail shown just below. This would provide additional information at no cost/visual overload.

The option to use the negative space of the y-axis is interesting too, but this is usually reserved to exports ([see here](https://energy-charts.info/charts/energy/chart.htm?l=en&c=FR&month=-1&legendItems=4x0vtu)).

I like @jayaddison's suggestion to get rid of `|<-`, few sources would not have it. 100% produced local electricity is actually rare – which is itself an interesting fact.

**Comment by Gianfilippo980** [comment]
This may be stupid, but I think that the graphs could be clearer if the sources were arranged on a volatility scale, so something that always provides the base load would appear as a flat ribbon, maybe with a couple indents, with more variable sources on top otherwise it gets hard to notice if a source is contributing a stable amount or if there is some variation.

**Comment by VIKTORVAV99** [comment]
> This may be stupid, but I think that the graphs could be clearer if the sources were arranged on a volatility scale, so something that always provides the base load would appear as a flat ribbon at the bottom, maybe with a couple indents, with more variable sources on top otherwise it gets hard to notice if a source is contributing a stable amount or if there is some variation.

That's not really what this issue is focused about but it's tracked here already: #7711

**Comment by madsnedergaard** [comment]
Thanks for providing feedback here :) 
We're picking this up again and will keep the ideas shared here in mind during the design and implementation 🙌
